### PR TITLE
Fixed failing fallback translations with disabled ThemeBundle

### LINKED ISSUE #6999: [Theme] Translations in admin stop working when disabled
When you disable the theme bundle (comment out the bundle loading in AppKernel) and clear cache, all admin translations break.

### PR DESCRIPTION
**Pull request by NeverResponse**
| Q               | A
| --------------- | ---
| Bug fix?        | yes |
| New feature?    | no |
| BC breaks?      | yes |
| Related tickets | fixes #6999, related #5031 |
| License         | MIT |

Todo:
* [x] Upgrade the upgrade file
* [x] Unit tests for the translator

### CODE REVIEW

File: src/Sylius/Bundle/ResourceBundle/Translation/Translator.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+final class Translator extends BaseTranslator
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected function computeFallbackLocales($locale)
+    {
+        $fallbackLocales = [];
+        foreach (array_merge([$locale], $this->getFallbackLocales()) as $fallbackLocale) {
+            foreach ($this->generateLocales($fallbackLocale) as $generatedLocale) {
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ResourceBundle/Translation/Translator.php]: Could be extracted to some private method

File: src/Sylius/Bundle/ResourceBundle/Resources/config/services.xml [hunk 2]
```diff
@@ -21,6 +21,7 @@
 
     <parameters>
         <parameter key="sylius.state_machine.class">Sylius\Component\Resource\StateMachine\StateMachine</parameter>
+        <parameter key="sylius.translator.class">Sylius\Bundle\ResourceBundle\Translation\Translato</parameter>
```
Comment (bigfoot90) [inline src/Sylius/Bundle/ResourceBundle/Resources/config/services.xml]: Should be `Sylius\Bundle\ResourceBundle\Translation\Translator`

Comment (NeverResponse) [inline src/Sylius/Bundle/ResourceBundle/Resources/config/services.xml]: Got lost while testing 😄

**Comment by NeverResponse** [comment]
A slight change in the suggested solution (which will basically remove all content from this PR 😄 ). 
Instead of introducing an overriding translator, which might quite possibly interfere with future symfony's BC breaks in their `Translator` class. We'll take a simplistic path. Adding a second `fallback` to the config's array (since symfony made that possible [it used to be a variable, not an array]), and simplify Theme's translator fallback resolving.

TL;DR
Scrapping custom translator to mostly use symfony's built-in translating solution.

File: src/Sylius/Bundle/ThemeBundle/Tests/Translation/TranslatorTest.php [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+use Sylius\Bundle\ThemeBundle\Translation\Translator;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Translation\Loader\LoaderInterface;
+use Symfony\Component\Translation\MessageCatalogue;
+use Symfony\Component\Translation\MessageSelector;
+
+/**
+ * @author Jan Góralski <jan.goralski@lakion.com>
```
Comment (pamil) [inline src/Sylius/Bundle/ThemeBundle/Tests/Translation/TranslatorTest.php]: Isn't it copied from the original test? `@see` annotation would be nice 🎉

File: src/Sylius/Bundle/ThemeBundle/Tests/Translation/TranslatorTest.php [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+ */
+class TranslatorTest extends \PHPUnit_Framework_TestCase
```
Comment (pamil) [inline src/Sylius/Bundle/ThemeBundle/Tests/Translation/TranslatorTest.php]: final? 🗡

File: src/Sylius/Bundle/ThemeBundle/Translation/Translator.php [hunk 5]
```diff
@@ -121,7 +112,38 @@ protected function initializeCatalogue($locale)
...
+
+        return array_unique($fallbackLocales);
+    }
+
+    /**
+     * @param string $locale
+     *
+     * @return string[]
+     */
+    private function getLocaleAndThemeSuffix($locale)
```
Comment (pamil) [inline src/Sylius/Bundle/ThemeBundle/Translation/Translator.php]: Can we extract it to `getLocaleWithoutSuffix` (suffix -> modifier, wdyt?) and `getLocaleSuffix` (suffix -> modifier, wdyt?)? Or just a simple, plain VO that gives us accessors for locale elements `[language[_territory][.codeset][@modifier]]` - `Locale::fromString('en_US.UTF-8@theme=heron')->getModifier()`?

File: src/Sylius/Bundle/ThemeBundle/Tests/Translation/TranslatorTest.php [hunk 6]
```diff
@@ -0,0 +1,343 @@
...
+
+    /**
+     * @test
+     * @dataProvider getAllValidLocalesTests
+     */
+    public function its_fallback_locales_can_be_set_only_if_valid($locale)
+    {
+        $translator = $this->createTranslator($locale);
+        $translator->setFallbackLocales(['fr', $locale]);
+        // no assertion. this method just asserts that no exception is thrown
```
Comment (Arminek) [inline src/Sylius/Bundle/ThemeBundle/Tests/Translation/TranslatorTest.php]: weird assumption.

File: UPGRADE.md [hunk 7]
```diff
@@ -166,6 +170,8 @@ These services will be generated automatically based on subject name.
 
 ### Configuration
 
+
```
Comment (lchrusciel) [inline UPGRADE.md]: Redundant

File: src/Sylius/Bundle/ThemeBundle/Tests/Translation/TranslatorTest.php [hunk 8]
```diff
@@ -0,0 +1,343 @@
...
+use Symfony\Component\Translation\Loader\LoaderInterface;
+use Symfony\Component\Translation\MessageCatalogue;
+use Symfony\Component\Translation\MessageSelector;
+
+/**
+ * @see \Symfony\Component\Translation\Tests\TranslatorTest
+ *
+ * @author Jan Góralski <jan.goralski@lakion.com>
+ */
+class TranslatorTest extends \PHPUnit_Framework_TestCase
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ThemeBundle/Tests/Translation/TranslatorTest.php]: final?

File: src/Sylius/Bundle/ThemeBundle/Tests/Translation/TranslatorTest.php [hunk 9]
```diff
@@ -0,0 +1,343 @@
...
+    /**
+     * @test
+     * @dataProvider getAllValidLocalesTests
+     */
+    public function it_gets_catalogue_with_fallback_catalogues_if_necessary($locale)
+    {
+        $translator = $this->createTranslator($locale);
+        $catalogue = new MessageCatalogue($locale);
+
+        if (false !== $themeDelimiter = strrpos($locale, '@')) {
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ThemeBundle/Tests/Translation/TranslatorTest.php]: if statements in test? Can't we test it separately?

**Comment by michalmarcinkowski** [comment]
@pamil does it look good?

**Comment by pjedrzejewski** [comment]
Thanks Jasiek!
